File: src/main/java/org/zzn/hospital/controllers/Appointmentcontroller.java
package org.zzn.hospital.controllers;

import lombok.AllArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.zzn.hospital.dtos.AppointmentDto;
import org.zzn.hospital.services.AppointmentService;

import java.util.List;

@RestController
@RequestMapping("/api/appointments")
@AllArgsConstructor
public class Appointmentcontroller {
    private final AppointmentService appointmentService;

    @PostMapping
    public ResponseEntity<AppointmentDto> create(@RequestBody AppointmentDto dto) {
        AppointmentDto created = appointmentService.create(dto);
        return new ResponseEntity<>(created, HttpStatus.CREATED);
    }

    @GetMapping("/{id}")
    public ResponseEntity<AppointmentDto> getById(@PathVariable Long id) {
        return ResponseEntity.ok(appointmentService.findById(id));
    }

    @GetMapping
    public ResponseEntity<List<AppointmentDto>> getAll() {
        return ResponseEntity.ok(appointmentService.findAll());
    }

}

File: src/main/java/org/zzn/hospital/Controllers/AppointmentController.kt
package org.zzn.hospital.Controllers

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*
import org.zzn.hospital.Models.Appointment
import org.zzn.hospital.Services.AppointmentService


@RestController
@RequestMapping("/appointments")
public class AppointmentController {
    @Autowired
    private val service: AppointmentService? = null

    @GetMapping
    fun getAll(): MutableList<Appointment?>? {
        return service!!.getAll()
    }

    @GetMapping("/{id}")
    fun getById(@PathVariable id: Long?): Appointment? {
        return service!!.getById(id)
    }

    @PostMapping
    fun create(@RequestBody appointment: Appointment?): Appointment? {
        return service!!.save(appointment)
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long?) {
        service!!.delete(id)
    }
}